docs/unittesting_examples.md

```diff
@@ -1,4 +1,318 @@
+Solidity introduced `try-catch` statement in version 0.6.0 which helps a lot to solve this purpose. Previously, this could be achieved using low-level call to some extent. Here is the example to test such a scenario:
+
+Contract/Program to be tested: `AttendanceRegister.sol`
+
+```
+pragma solidity >=0.4.22 <0.7.0;
+contract AttendanceRegister {
+struct Student{
+        string name;
+        uint class;
+    }
+event Added(string name, uint class, uint time);
+mapping(uint => Student) public register; // roll number => student details
+    function add(uint rollNumber, string memory name, uint class) public returns (uint256){
+        require(class > 0 && class <= 12, "Invalid class");
+        require(register[rollNumber].class == 0, "Roll number not available");
+        Student memory s = Student(name, class);
+        register[rollNumber] = s;
+        emit Added(name, class, now);
+        return rollNumber;
+    }
+    
+    function getStudentName(uint rollNumber) public view returns (string memory) {
+        return register[rollNumber].name;
+    }
+}
+```
+
+Test contract/program: `AttendanceRegister_test.sol`
+
+```
+pragma solidity >=0.4.22 <0.7.0;
+import "remix_tests.sol"; // this import is automatically injected by Remix.
+import "./AttendanceRegister.sol";
+contract AttendanceRegisterTest {
+   
+    AttendanceRegister ar;
+    
+    /// 'beforeAll' runs before all other tests
+    function beforeAll () public {
+        // Create an instance of contract to be tested
+        ar = new AttendanceRegister();
+    }
+    
+    /// For solidity version greater or equal to 0.6.0, 
+    /// See: https://solidity.readthedocs.io/en/v0.6.0/control-structures.html#try-catch
+    /// Test 'add' using try-catch
+    function testAddSuccessUsingTryCatch() public {
+        // This will pass
+        try ar.add(101, 'secondStudent', 11) returns (uint256 r) {
+            Assert.equal(r, 101, 'wrong rollNumber');
+        } catch Error(string memory /*reason*/) {
+            // This is executed in case
+            // revert was called inside getData
+            // and a reason string was provided.
+            Assert.ok(false, 'failed with reason');
+        } catch (bytes memory /*lowLevelData*/) {
+            // This is executed in case revert() was used
+            // or there was a failing assertion, division
+            // by zero, etc. inside getData.
+            Assert.ok(false, 'failed unexpected');
+        }
+    }
+    
+    /// Test failure case of 'add' using try-catch
+    function testAddFailureUsingTryCatch1() public {
+        // This will revert on 'require(class > 0 && class <= 12, "Invalid class");' for class '13'
+        try ar.add(101, 'secondStudent', 13) returns (uint256 r) {
+            Assert.ok(false, 'method execution should fail');
```

Comment:
I'll ask you about this line tomorrow - about seeing the 'method execution should fail'

Reply:
That Assert is symbolic. That line will be never executed with `ar.add(101, 'secondStudent', 13)` throwing error. If due to any mistake in providing input for fail case, it is somehow passed then that `Assert` will fail stating `method execution should fail`.
Let me know your thoughts.